src/backend/gl/src/lib.rs

```diff
@@ -528,115 +528,116 @@ impl CommandQueue {
-                                base_vertex as gl::types::GLint,
-                                base_instance as gl::types::GLuint,
                             );
-                        } else if base_instance != 0 {
-                            error!("Instance bases with instanced indexed drawing is not supported")
-                        } else {
-                            // No base instance
-                            if features.draw_indexed_instanced_base_vertex {
-                                gl.DrawElementsInstancedBaseVertex(
-                                    primitive,
-                                    count as gl::types::GLsizei,
-                                    index_type,
-                                    offset,
-                                    num as gl::types::GLsizei,
-                                    base_vertex as gl::types::GLint,
-                                );
-                            } else if base_vertex != 0 {
-                                error!("Base vertex with instanced indexed drawing is not supported")
-                            } else {
-                                // No base instance and base vertex
-                                if features.draw_indexed_instanced {
-                                    gl.DrawElementsInstanced(
-                                        primitive,
-                                        count as gl::types::GLsizei,
-                                        index_type,
-                                        offset,
-                                        num as gl::types::GLsizei,
-                                    );
-                                } else {
-                                    error!("Instanced indexed drawing is not supported")
-                                }
-                            }
                         }
-                    },
-                    None => unsafe {
-                        if features.draw_indexed_base {
+                    } else if features.draw_indexed_base {
+                        unsafe {
                             gl.DrawElementsBaseVertex(
                                 primitive,
-                                count as gl::types::GLsizei,
+                                index_count as _,
                                 index_type,
                                 offset,
-                                base_vertex as gl::types::GLint,
+                                base_vertex as _,
                             );
-                        } else if base_vertex == 0 {
-                            gl.DrawElements(
+                        }
+                    } else {
+                        error!("Base vertex with indexed drawing not supported");
+                    }
+                } else if features.draw_indexed_instanced {
+                    if base_vertex == 0 && instances.start == 0 {
```

Comment:
Can we reduce the checks for `instances.start == 0`? Currently 3 times in row

Reply:
it would require more nesting/depth to the code structure, I'd rather have this condition repeated instead